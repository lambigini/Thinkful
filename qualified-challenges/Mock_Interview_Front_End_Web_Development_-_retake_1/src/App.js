import React, { useEffect, useState } from "react";
import AlbumDetail from "./AlbumDetail";

function App() {
  const [albums, setAlbums] = useState([]);

  useEffect(() => {
    fetch("https://jsonplaceholder.typicode.com/albums?userId=1")
      .then((response) => response.json())
      .then(setAlbums)
      .catch((error) => {
        console.log(error);
      });
  }, []);

  // render title
  // album title, map

  // onclik handle, new photo
  // map show the new photo

  const [photo, setPhoto] = useState([]);

  const handleClick = (albumId) => {
    // console.log("clicked");

    fetch(`https://jsonplaceholder.typicode.com/albums/${albumId}/photos`)
      .then((response) => response.json())
      .then(setPhoto)
      .catch((error) => {
        console.log(error);
      });
  };

  // console.log("photo, ", photo);

  const photourl = photo.slice(0, 10).map((p, index) => {
    // if (index < 11) {
    return (
      <li key={index}>
        {p.title}
        <img
          // src={p.thumbnailUrl}
          src={p.url}
          alt={p.title}
        />
      </li>
    );
    // }
  });

  const title = albums.map((album) => (
    <li key={album.id} onClick={() => handleClick(album.id)}>
      {album.title}
    </li>
  ));

  return (
    <div className="App">
      <ul>{title}</ul>
      <ul> {photourl}</ul>
    </div>
  );
}

export default App;
