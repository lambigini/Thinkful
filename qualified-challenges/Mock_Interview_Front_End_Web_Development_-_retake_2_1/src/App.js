import React, { useEffect, useState } from "react";
import UserTodos from "./UserTodos.js";

function App() {
  const [users, setUsers] = useState([]);

  useEffect(() => {
    fetch("https://jsonplaceholder.typicode.com/users")
      .then((response) => response.json())
      .then(setUsers)
      .catch((error) => {
        console.log(error);
      });
  }, []);

  const [todos, setTodos] = useState([]);

  const handleClick = (id) => {
    // console.log("click");
    fetch(`https://jsonplaceholder.typicode.com/users/${id}/todos`)
      .then((response) => response.json())
      .then(setTodos)
      .catch((error) => {
        console.log(error);
      });
  };

  const listName = users.map((user, index) => (
    <li key={user.id} onClick={() => handleClick(user.id)}>
      {user.name}
    </li>
  ));

  const showTodo = todos.map((todo) => <li>{todo.title}</li>);

  return (
    <div className="App">
      {/* Write your code below */}
      <ul>{listName}</ul>
      <ul>{showTodo}</ul>
    </div>
  );
}

export default App;
