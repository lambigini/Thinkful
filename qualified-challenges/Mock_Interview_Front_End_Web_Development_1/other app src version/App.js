// import React, { useEffect, useState } from "react";
// import Comments from "./Comments";
// import PostDetail from "./PostDetail";

// function App() {
//   const [posts, setPosts] = useState([]);

//   useEffect(() => {
//     fetch("https://jsonplaceholder.typicode.com/todos?userId=1")
//       .then((response) => response.json())
//       .then(setPosts)
//       .catch((error) => {
//         console.log(error);
//       });
//   }, []);

//   console.log("post,", posts);
//   const handleDelete = (post) => {
//     console.log("click", post);

//     post.completed = !post.completed;
//     console.log("after click ", post);
//     setPosts((current) => current.map((post1, index) => post1));
//   };

//   const title = posts.map((post, index) => (
//     <li
//       key={post.id}
//       onClick={() => handleDelete(post)}
//       style={{ textDecoration: post.completed ? "line-through" : "" }}
//     >
//       {post.title}
//     </li>
//   ));

//   return (
//     <div>
//       <ol>{title}</ol>
//     </div>
//   );
// }

// export default App;

import React, { useEffect, useState } from "react";
import Comments from "./Comments";
import PostDetail from "./PostDetail";

function App() {
  const [posts, setPosts] = useState([]);

  useEffect(() => {
    fetch("https://jsonplaceholder.typicode.com/posts?userId=1")
      .then((response) => response.json())
      .then(setPosts)
      .catch((error) => {
        console.log(error);
      });
  }, []);

  const [newPost, setNewPost] = useState([]);

  const handleClick = (post) => {
    console.log("post", post);
    fetch(`https://jsonplaceholder.typicode.com/posts/${post.id}/comments`)
      .then((response) => response.json())
      .then(setNewPost);

    console.log("newPost", newPost);
  };

  const postObject = posts.map((post, index) => {
    return (
      <PostDetail
        post={post}
        key={index}
        handleClick={() => handleClick(post)}
      />
    );
  });

  // if (newPost.length > 1) {
  //   const comments = newPost.map((post, index) => {
  //     return <Comments post={post} key={index} />;
  //   });

  //   return (
  //     <div className="App">
  //       {postObject}
  //       {comments}
  //     </div>
  //   );
  // } else {
  //   return <div className="App">{postObject}</div>;
  // }

  const comments = newPost.map((post, index) => {
    return <Comments post={post} key={index} />;
  });

  return (
    <div className="App">
      {postObject}
      {comments}
    </div>
  );
}

export default App;
