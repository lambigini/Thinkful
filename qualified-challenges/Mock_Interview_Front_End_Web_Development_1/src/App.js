import React, { useEffect, useState } from "react";
import Comments from "./Comments";
import PostDetail from "./PostDetail";

function App() {
  const [posts, setPosts] = useState([]);

  useEffect(() => {
    fetch("https://jsonplaceholder.typicode.com/posts?userId=1")
      .then((response) => response.json())
      .then(setPosts)
      .catch((error) => {
        console.log(error);
      });
  }, []);

  const [newPost, setNewPost] = useState([]);

  const handleClick = (post) => {
    console.log("post", post);
    fetch(`https://jsonplaceholder.typicode.com/posts/${post.id}/comments`)
      .then((response) => response.json())
      .then(setNewPost);

    console.log("newPost", newPost);
  };

  const postObject = posts.map((post, index) => {
    return (
      <PostDetail
        post={post}
        key={index}
        handleClick={() => handleClick(post)}
      />
    );
  });

  if (newPost.length > 1) {
    const comments = newPost.map((post, index) => {
      return <Comments post={post} key={index} />;
    });

    return (
      <div className="App">
        {postObject}
        {comments}
      </div>
    );
  } else {
    return <div className="App">{postObject}</div>;
  }
}

export default App;
