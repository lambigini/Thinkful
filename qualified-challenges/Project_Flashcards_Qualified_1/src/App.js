import React from "react";
import { Route, Switch } from "react-router-dom";
import Layout from "./Layout";
import NotFound from "./Layout/NotFound";
import ViewAllDesk from "./Layout/Desk/ViewAllDesk";
import Study from "./Layout/Desk/Study";
/**
 * App is a wrapper for <Layout>, you should not need to change this file.
 */

function App() {
  return (
    <div className="app-routes">
      <Switch>
        <Route exact path="/">
          <Layout />
        </Route>

        <Route exact path="/decks/:deckId">
          <ViewAllDesk />
        </Route>

        <Route path="/decks/:deckId/study">
          <Study />
        </Route>

        <Route>
          <NotFound />
        </Route>
      </Switch>
    </div>
  );
}

export default App;
