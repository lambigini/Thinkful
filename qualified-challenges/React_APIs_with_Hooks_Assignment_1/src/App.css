.App {
  display: flex;
}

.column {
  padding-right: 1em;
}

ul {
  list-style: none;
  padding-left: 16px;
}

li {
  margin: 0.5em;
}

button {
  width: 200px;
}
