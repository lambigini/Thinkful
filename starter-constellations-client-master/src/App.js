import React, { useEffect, useState } from "react";
import Header from "./common/Header";
import Navigation from "./common/Navigation";
import ConstellationsList from "./home/ConstellationsList";

function App() {
  const [constellations, setConstellations] = useState({
    all: [],
    visible: [],
  });

  useEffect(() => {
    const url = `${process.env.REACT_APP_API_BASE_URL}/constellations`;
    fetch(url)
      .then((response) => response.json())
      .then((response) => {
        setConstellations({
          all: response,
          visible: response,
        });
      });
  }, []);

  return (
    <main>
      <Header
        constellations={constellations}
        setConstellations={setConstellations}
      />
      <Navigation />
      <ConstellationsList constellations={constellations.visible} />
    </main>
  );
}

export default App;
