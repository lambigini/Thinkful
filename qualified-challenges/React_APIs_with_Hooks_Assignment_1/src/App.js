import React, { useState, useEffect } from "react";
import "./App.css";

import AlbumList from "./AlbumList";
import UserList from "./UserList";

function App() {
  const [users, setUsers] = useState([]);
  const [currentUser, setCurrentUser] = useState({});
  const [albums, setAlbums] = useState([]);
  const oldTitle = document.title;

  useEffect(() => {
    const abortController = new AbortController();
    document.title = "Awesome Album App";
    setUsers([]);
    async function loadUsers() {
      try {
        const response = await fetch(
          `https://jsonplaceholder.typicode.com/users`,
          { signal: abortController.signal }
        );
        const userFromAPI = await response.json();
        setUsers(userFromAPI);
      } catch (error) {
        if (error.name === "AbortError") {
        } else {
          throw error;
        }
        console.log("Users error", error);
      }
    }

    loadUsers();

    return () => {
      abortController.abort();
      document.title = oldTitle;
    };
  }, []);

  useEffect(() => {
    const abortController = new AbortController();
    setAlbums([]);
    async function loadAlbums() {
      try {
        const response = await fetch(
          `https://jsonplaceholder.typicode.com/albums?userId=${currentUser.id}`,

          { signal: abortController.signal }
        );

        const albumsFromAPI = await response.json();

        setAlbums(albumsFromAPI);
      } catch (error) {
        //include this block of code will made the test to failed
        // if (error.name === "AbortError") {
        //   console.log("Aborted, ", user.id);
        // }
        // else {
        //   throw error;
        // }
        console.log("Album error", error);
      }
    }

    loadAlbums();

    return () => abortController.abort();
  }, [currentUser]);
  if (users.length > 0) {
    return (
      <div className="App">
        <div className="left column">
          <UserList users={users} setCurrentUser={setCurrentUser} />
        </div>
        <div className="right column">
          <AlbumList user={currentUser} albums={albums} />
        </div>
      </div>
    );
  } else {
    return <p>loading...</p>;
  }
}

export default App;
