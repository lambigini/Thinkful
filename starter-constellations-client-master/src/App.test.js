import React from "react";
import App from "./App";
import renderer from "react-test-renderer";

it("renders the NoConstellationsMessage when no data appears", () => {
  const component = renderer.create(<App />);
  expect(component.toJSON()).toMatchSnapshot();
});
