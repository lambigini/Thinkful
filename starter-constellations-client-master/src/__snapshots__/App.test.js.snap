// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the NoConstellationsMessage when no data appears 1`] = `
<main>
  <div
    className="jumbotron jumbotron-fluid bg-info text-white rounded-0 mb-0"
  >
    <div
      className="container"
    >
      <h1
        className="display-4"
      >
        I love Constellations
      </h1>
      <p
        className="lead"
      >
        Explore different constellations in the sky!
      </p>
      <hr />
      <form
        className="form container"
      >
        <div
          className="form-group row"
        >
          <input
            className="form-control col col-md-8"
            id="search"
            name="search"
            onChange={[Function]}
            placeholder="Search the stars..."
            type="text"
          />
          <label
            className="form-check-label sr-only"
            htmlFor="search"
          >
            Search
          </label>
        </div>
      </form>
    </div>
  </div>
  <ul
    className="nav justify-content-center bg-light py-2"
  >
    <li
      className="nav-item"
    >
      <button
        className="btn btn-link nav-link active"
      >
        Home
      </button>
    </li>
    <li
      className="nav-item"
    >
      <button
        className="btn btn-link nav-link"
      >
        About
      </button>
    </li>
  </ul>
  <div
    className="container my-5"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <p
          className="text-center"
        >
          There are no constellations listed...
        </p>
      </div>
    </div>
  </div>
</main>
`;
